### PR TITLE
HDX-9752 Adding data access layer for patch table

### PR DESCRIPTION
**Pull request by IanHopkinson**
This PR introduces some functions to query the patch table

Currently tests are passing locally but failing in GitHub Actions due to an integrity violation for the id field which should be autoincrementing

### CODE REVIEW

**Comment by IanHopkinson** [comment]
I fixed the issue with a test failing in GitHub Actions but not locally with help from @alexandru-m-g - the behaviour has changed slightly. When you do an insert, the function returns the id of the inserted object.

I think this is ready now.

File: tests/conftest.py [hunk 1]
```diff
@@ -32,22 +39,20 @@ def session_maker() -> sessionmaker[Session]:
...
         return [row[0] for row in result if row != 'alembic_version']
     except Exception as e:
         raise e
     finally:
         session.close()
 
 
-@pytest.fixture(scope='function')
-def clear_db_tables(log: Logger, session_maker: sessionmaker[Session], list_of_db_tables: List[str]):
+@pytest.fixture(scope='session')
```
Comment (alexandru-m-g) [inline tests/conftest.py:54]: @IanHopkinson , can you give more details on the change to the _scope_ of the fixture ? With "session" doesn't it mean that the db will be cleaned only once per testing session ? I was thinking that each test would expect to have a clean db.

Comment (IanHopkinson) [reply tests/conftest.py:54]: I tried making this "function" level but the tests seem to take forever to run, I'm trying to diagnose why

Comment (IanHopkinson) [reply tests/conftest.py]: Thanks for your fix @alexandru-m-g - the tests now pass and the fixtures are refreshed at the function level.

I believe this is ready to merge now.

**Pull request merged**